### PR TITLE
profile and background image cleanup from cloud

### LINKED ISSUE #1079: Cleanup of profile image
### Description

When user changes their profile image its ref is deleted from the user collection but the image is still available in the cloud storage. And the same goes for background image I believe. I want to cleanup both of them.

### Steps to reproduce

Check the function for changing image. It just update the photoURL and nothing else.

### Screenshots

_No response_

### Desktop (please complete the following information)

_No response_

### Smartphone (please complete the following information)

_No response_

### Additional information

_No response_

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I have checked the existing issues
- [X] I'm a GSSoC'23 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by Palakkgoyal**
This PR closes #1079 
I have done the cleanup for both and also made the function reusable.

### CODE REVIEW

File: src/components/EditProfile/index.jsx [hunk 1]
```diff
@@ -78,7 +80,8 @@ const EditProfile = ({ userData, username, setIsEditing, setUserData }) => {
     if (!usernameAvailable) {
       return;
     }
-    if (image) {
+    const isImgDeleted = deleteImg(userData.avatar)
```
Comment (narayan954) [inline src/components/EditProfile/index.jsx:83]: isn't that an async call and should be awaited? Also, we should be deleting the image only after new one replaces it. you may keep this url in variable and call it after update is successful to free the resources

File: src/pages/Profile/index.jsx [hunk 2]
```diff
@@ -80,7 +81,8 @@ function Profile() {
 
   const handleBgImageSave = () => {
     try {
-      if (backgroundImage) {
+      const imgdeleted = deleteImg(bgImageUrl)
```
Comment (narayan954) [inline src/pages/Profile/index.jsx:84]: same suggestion as before

**Review by narayan954 — CHANGES_REQUESTED** [review]
some changes required

**Comment by Palakkgoyal** [comment]
Yes, I was confused about it that I should delete the image before or after.

**Comment by Palakkgoyal** [comment]
Hey @narayan954 do you know why this error is keep coming?
Screenshot 2023-08-04 140540

**Comment by narayan954** [comment]
> Hey @narayan954 do you know why this error is keep coming? Screenshot 2023-08-04 140540

need to look into this, step to reproduce this error?

**Comment by Palakkgoyal** [comment]
> > Hey @narayan954 do you know why this error is keep coming? Screenshot 2023-08-04 140540
> 
> need to look into this, step to reproduce this error?
> 
> Edit: Fixed this, default firebase rules were not updated

Ohk, thanks.

**Comment by Palakkgoyal** [comment]
I did the required changes. Can you please check?

**Review by narayan954 — APPROVED** [review]
looks good to me!

**Pull request merged**